File: Stage19/Python/Problem11729.py
import sys
input = sys.stdin.readline

def hanoi(n, start, mid, end, result):
    if n == 1:
        result.append(f"{start} {end}")
        return
    hanoi(n - 1, start, end, mid, result)
    result.append(f"{start} {end}")
    hanoi(n - 1, mid, start, end, result)

def main():
    n = int(input())
    result = []
    hanoi(n, 1, 2, 3, result)
    print(len(result))
    print("\n".join(result))

if __name__ == "__main__":
    main()

File: Stage19/C/Problem11729.c
#include <stdio.h>

void hanoi(int n, int start, int mid, int end) {
    if (n == 1) {
        printf("%d %d\n", start, end);
        return;
    }
    hanoi(n - 1, start, end, mid);
    printf("%d %d\n", start, end);
    hanoi(n - 1, mid, start, end);
}

int main(void) {
    int N;
    scanf("%d", &N);

    printf("%d\n", (1 << N) - 1);

    hanoi(N, 1, 2, 3);

    return 0;
}